### PR TITLE
[GOBBLIN-1038] Set default dataset descriptor configs based on the DataNode

### PR DESCRIPTION
**Pull request by jack-moseley**
Dear Gobblin maintainers,

Please accept this PR. I understand that it will not be reviewed until I have checked off all the steps below!

### JIRA
- [x] My PR addresses the following [Gobblin JIRA](https://issues.apache.org/jira/browse/GOBBLIN/) issues and references them in the PR title. For example, "[GOBBLIN-XXX] My Gobblin PR"
    - https://issues.apache.org/jira/browse/GOBBLIN-1038

### Description
- [x] Here are some details about my PR, including screenshots (if applicable):

Set default dataset descriptor config for class/platform by including them in each DataNode. For ones like `BaseDataNode` where it doesn't make sense, they are null so there is no default.

Also added a check that all dest nodes are the same type of datanode since it wouldn't work anyway for there to be multiple types.

## Tests
- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits
- [x] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

### CODE REVIEW

**Comment by jack-moseley** [comment]
@arjun4084346 please review

File: gobblin-service/src/main/java/org/apache/gobblin/service/modules/flowgraph/pathfinder/AbstractPathFinder.java [hunk 1]
```diff
@@ -103,6 +105,11 @@
       this.destNodes.add(destNode);
     }
 
+    // All dest nodes should be the same class
+    if (this.destNodes != null && this.destNodes.stream().map(Object::getClass).collect(Collectors.toSet()).size() > 1) {
```
Comment (arjun4084346) [inline gobblin-service/src/main/java/org/apache/gobblin/service/modules/flowgraph/pathfinder/AbstractPathFinder.java]: Can our path finding algorithm return dest nodes with different types? just curious.

**Comment by arjun4084346** [comment]
+1 LGTM.

File: gobblin-service/src/main/java/org/apache/gobblin/service/modules/flowgraph/pathfinder/AbstractPathFinder.java (hunk 1 above)
Comment (jack-moseley) [reply gobblin-service/src/main/java/org/apache/gobblin/service/modules/flowgraph/pathfinder/AbstractPathFinder.java]: This is the destination nodes specified by the user in the `destinationIdentifiers` property, it allows more than one. This makes it explicitly fail if they specified an HDFS and mysql cluster for example (which would probably fail somewhere else in compilation anyway).

**Pull request closed without merging**